src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts

```diff
@@ -147,9 +147,15 @@ export class CheckboxSelectColumn<T extends Slick.SlickData> implements Slick.Pl
 	}
 
 	private handleHeaderClick(e: Event, args?: Slick.OnHeaderClickEventArgs<T>): void {
-		this.onHeaderCheckboxStateChange();
-		e.preventDefault();
-		e.stopPropagation();
+		if (this.isCheckAllHeaderCheckboxShown()) {
```

Comment:
You should only be attaching a click handler if there's a checkbox - otherwise screen readers will announce "clickable" on the header even if there's nothing to click. 

(that might require a change in slickgrid, but at the very least this code can be simplified by only subscribing to the event if there's a checkbox)

Reply:
That doesn't seem to be a problem as the headers are not focusable right now. I am now subscribing to the header event only when check all box is shown.